### PR TITLE
Force nightly pyarrow in the upstream build

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Similar as https://github.com/dask/dask/pull/8281, it's still not fully clear why it is not automatically picking up the most recent version

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
It seems the test is hanging after `test_modification_time_read_bytes`, but so if I look at the log output on main where it succeeds, the next test is:

```
dask/bytes/tests/test_s3.py::test_modification_time_read_bytes PASSED    [ 41%]
dask/bytes/tests/test_s3.py::test_parquet[True-pyarrow] PASSED           [ 41%]
```

So which _is_ a pyarrow related test, and so it might actually be identifying an issue with the latest dask / pyarrow combo.

**Comment by jorisvandenbossche** [comment]
So I can reproduce this locally:

```
$ pytest dask/bytes/tests/test_s3.py::test_parquet[True-pyarrow] -vvv -s
...
dask/bytes/tests/test_s3.py::test_parquet[True-pyarrow]  * Running on http://127.0.0.1:5555 (Press CTRL+C to quit)
127.0.0.1 - - [29/Apr/2022 10:42:19] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test/test/accounts.1.json HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test/test/accounts.2.json HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test/test.parquet/part.0.parquet HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test/test.parquet/part.1.parquet HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test/test.parquet/_common_metadata HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "PUT /test/test.parquet/_metadata HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "GET /test?list-type=2&prefix=test.parquet%2F&delimiter=%2F&encoding-type=url HTTP/1.1" 200 -
127.0.0.1 - - [29/Apr/2022 10:42:19] "GET /test?list-type=2&prefix=test.parquet%2F&delimiter=%2F&encoding-type=url HTTP/1.1" 200 -
```

The test is hanging with the above output, and I can't even interrupt it (with ctrl-c) but have to close the terminal.

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jorisvandenbossche for updating the CI environment and debugging this issue. Should we temporarily skip the hanging test and merge this PR in?

**Comment by jorisvandenbossche** [comment]
I have a PR open to fix this (https://github.com/apache/arrow/pull/13033), so we can probably wait with merging this PR until it is fixed. But in the meantime, I did add temporary skips, so we can at least check the rest of the tests on this PR.

**Comment by jorisvandenbossche** [comment]
OK, the test build is now finishing. There are still some failures, because of a deprecation warning for `ParquetDataset.metadata` attribute that gets turned into an error when running tests.

So that means that the pyarrow.dataset engine is still using the legacy ParquetDataset API in some place (xref https://github.com/dask/dask/issues/8243). cc @rjzamora

**Comment by jorisvandenbossche** [comment]
Ah, it seems this is only done in a helper function defined in the tests itself:

https://github.com/dask/dask/blob/4d6a5f08c45be56302f696ca4ef6038a1cd1e734/dask/dataframe/io/tests/test_parquet.py#L1735-L1746

That should be possible to rewrite to use `pq.read_metadata` instead

**Comment by jcrist** [comment]
> Ah, it seems this is only done in a helper function defined in the tests itself:

Nice catch, I'll push a fix up for this.

> I have a PR open to fix this (https://github.com/apache/arrow/pull/13033), so we can probably wait with merging this PR until it is fixed. But in the meantime, I did add temporary skips, so we can at least check the rest of the tests on this PR.

Thanks Joris! We might have had a user run into this issue last week (never determined if it was pyarrow or fsspec's fault). Hopefully this fixed their problem too :crossed_fingers:.

**Comment by jorisvandenbossche** [comment]
> Nice catch, I'll push a fix up for this.

Hopefully you didn't start on that yet, as I already included a commit here as well

**Comment by jcrist** [comment]
I didn't, thanks for letting me know :)

**Review by jcrist — APPROVED** [review]
LGTM!

File: dask/dataframe/io/tests/test_parquet.py [hunk 1]
```diff
@@ -1662,7 +1662,7 @@ def check_compression(engine, filename, compression):
         else:
             assert md.total_compressed_size != md.total_uncompressed_size
     else:
-        metadata = pa.parquet.ParquetDataset(filename).metadata
+        metadata = pa.parquet.read_metadata(filename + "/_metadata")
```
Comment (jrbourbeau) [inline dask/dataframe/io/tests/test_parquet.py:1665]: I thought we would have needed to use `os.path.join(...)` here for this to work on Windows, but I guess not...It looks like tests that use `check_compression` are passing in our Windows builds.

**Review by jrbourbeau — COMMENTED** [review]
Thanks @jorisvandenbossche! 

> we can probably wait with merging this PR until it is fixed

So after removing the skips and rerunning CI, say tomorrow, after there is a new nightly then this should be good to go

File: dask/dataframe/io/tests/test_parquet.py (hunk 1 above)
Comment (jorisvandenbossche) [reply dask/dataframe/io/tests/test_parquet.py:1665]: But let's maybe make it more robust using os.path.join anyway.

**Comment by jorisvandenbossche** [comment]
It's still picking up the nightly package of yesterday. We had a failure that caused a few packages not being uploaded, among which the linux one for py3.9. So will have to retry tomorrow.

**Comment by jorisvandenbossche** [comment]
This is finally passing now!

**Review by jrbourbeau — APPROVED** [review]
Hooray -- thanks @jorisvandenbossche!

**Pull request merged**